xapian-letor/tests/harness/testutils.h

```diff
@@ -90,4 +90,11 @@ void test_mset_order_equal(const Xapian::MSet &mset1,
 	expected_exception = NULL;\
     } while (0)
 
+/// Check that CODE throws exactly Xapian LetorParseError TYPE.
+#define TEST_PARSE_EXCEPTION(db_path, qrel, data_directory, testfile)\
+	TEST_EXCEPTION(Xapian::LetorParseError,\
+		       Xapian::prepare_training_file(db_path, data_directory +\
+						     testfile, qrel, 10,\
+						     "training_output.txt"));
```

Comment:
It's better not to include the `;` on the end of the definition of a function-like macro like this where a `;` is expected to be added when the macro is used.  People will tend to add a `;` without thinking about it, editors which auto-indent will do the right thing, etc.

In simple cases, the extra `;` you get if the macro includes one and there's also one where it is used makes no difference, but it can - e.g. consider:

```
if (foo)
    TEST_PARSE_EXCEPTION(db_path, qrel, data_directory, "foo.txt");
else
    TEST_PARSE_EXCEPTION(db_path, qrel, data_directory, "bar.txt");
```

The first `TEST_PARSE_EXCEPTION` would currently expand to give a double `;` which means the `if` statement ends there and is followed by an empty statement, and then you'll get `error: ‘else’ without a previous ‘if’`` or similar for the `else`.

I think there are also situations where this can even result in a successful compilation which does the wrong thing but an example doesn't come to mind right now.

Reply:
fixed in a fixup commit.